### ISSUE TITLE
Doesn't package on linux

### ISSUE THREAD

**Issue by nukesparrow** [drrb/rust-netbeans#6]
Maven output:

$ mvn -e package
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Rust NetBeans Plugin 1.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- antlr4-maven-plugin:4.0:antlr4 (default) @ rust-netbeans ---
[INFO] ANTLR 4: Processing source directory /home/portable/tmp/rust-netbeans/src/main/antlr4
[INFO] Processing grammar: com/github/drrb/rust/netbeans/parsing/Rust.g4
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.719s
[INFO] Finished at: Fri Nov 21 12:15:46 YEKT 2014
[INFO] Final Memory: 14M/444M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.antlr:antlr4-maven-plugin:4.0:antlr4 (default) on project rust-netbeans: Execution default of goal org.antlr:antlr4-maven-plugin:4.0:antlr4 failed: Index: 0, Size: 0 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.antlr:antlr4-maven-plugin:4.0:antlr4 (default) on project rust-netbeans: Execution default of goal org.antlr:antlr4-maven-plugin:4.0:antlr4 failed: Index: 0, Size: 0
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:225)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:320)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal org.antlr:antlr4-maven-plugin:4.0:antlr4 failed: Index: 0, Size: 0
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:110)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
	... 19 more
Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:638)
	at java.util.ArrayList.get(ArrayList.java:414)
	at org.antlr.v4.misc.OrderedHashMap.getElement(OrderedHashMap.java:46)
	at org.antlr.v4.analysis.LeftRecursiveRuleTransformer.setAltASTPointers(LeftRecursiveRuleTransformer.java:241)
	at org.antlr.v4.analysis.LeftRecursiveRuleTransformer.translateLeftRecursiveRule(LeftRecursiveRuleTransformer.java:162)
	at org.antlr.v4.analysis.LeftRecursiveRuleTransformer.translateLeftRecursiveRules(LeftRecursiveRuleTransformer.java:89)
	at org.antlr.v4.semantics.SemanticPipeline.process(SemanticPipeline.java:94)
	at org.antlr.v4.Tool.processNonCombinedGrammar(Tool.java:399)
	at org.antlr.v4.Tool.process(Tool.java:384)
	at org.antlr.mojo.antlr4.Antlr4Mojo$CustomTool.process(Antlr4Mojo.java:461)
	at org.antlr.v4.Tool.processGrammarsOnCommandLine(Tool.java:343)
	at org.antlr.mojo.antlr4.Antlr4Mojo.execute(Antlr4Mojo.java:272)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
	... 20 more
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

**Comment by vbmacher** [comment]
When used `antlr4-maven-plugin` version 4.3 (current version), errors are different. I used cygwin for the compilation.

```
$ mvn install
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building Rust NetBeans Plugin 1.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- antlr4-maven-plugin:4.3:antlr4 (default) @ rust-netbeans ---
[INFO] ANTLR 4: Processing source directory D:\projects\rust-netbeans-master\src\main\antlr4
[ERROR] error(50): D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4:42:7: syntax error: '"' came as a complete surprise to me
[ERROR] D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4 [42:7]: syntax error: '"' came as a complete surprise to me
[ERROR] error(50): D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4:42:16: syntax error: '"' came as a complete surprise to me
[ERROR] D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4 [42:16]: syntax error: '"' came as a complete surprise to me
[ERROR] error(50): D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4:42:20: syntax error: '"' came as a complete surprise to me
[ERROR] D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4 [42:20]: syntax error: '"' came as a complete surprise to me
[ERROR] error(50): D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4:42:28: syntax error: '"' came as a complete surprise to me
[ERROR] D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4 [42:28]: syntax error: '"' came as a complete surprise to me
[ERROR] error(50): D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4:366:4: syntax error: '_' came as a complete surprise to me
[ERROR] D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4 [366:4]: syntax error: '_' came as a complete surprise to me
[ERROR] error(50): D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4:366:5: syntax error: '_' came as a complete surprise to me
[ERROR] D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4 [366:5]: syntax error: '_' came as a complete surprise to me
[ERROR] error(50): D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4:447:4: syntax error: '_' came as a complete surprise to me
[ERROR] D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4 [447:4]: syntax error: '_' came as a complete surprise to me
[ERROR] error(50): D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4:447:5: syntax error: '_' came as a complete surprise to me
[ERROR] D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4 [447:5]: syntax error: '_' came as a complete surprise to me
[ERROR] error(50): D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4:596:4: syntax error: '_' came as a complete surprise to me
[ERROR] D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4 [596:4]: syntax error: '_' came as a complete surprise to me
[ERROR] error(50): D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4:596:5: syntax error: '_' came as a complete surprise to me
[ERROR] D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4 [596:5]: syntax error: '_' came as a complete surprise to me
[ERROR] error(50): D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4:699:0: syntax error: '_' came as a complete surprise to me
[ERROR] D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4 [699:0]: syntax error: '_' came as a complete surprise to me
[ERROR] error(50): D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4:699:1: syntax error: '_' came as a complete surprise to me
[ERROR] D:\projects\rust-netbeans-master\src\main\antlr4\com\github\drrb\rust\netbeans\parsing\Rust.g4 [699:1]: syntax error: '_' came as a complete surprise to me
```

**Comment by drrb** [comment]
Hi All, thanks for the investigation and sorry about the delay. I've been in the process of replacing the ANTLR-based lexer/parser with a native one that uses the Rust compiler directly. It's in master now, and works well on OSX (the main OS I develop on). It also compiles properly on Linux (a step forward, at least!). Unfortunately it doesn't seem to load the native component on Linux properly (see #7). I'll keep investigating, and aiming to get it running on Linux. If anyone is able to work out why the native component is not loading properly on Linux, I'd love to hear from you! Meanwhile, I'll close this issue (the ANTLR issue is solved) and track any progress on #7.
